File: backend/src/main/java/edu/cit/cabasag/barberconnect/feature/social/Reaction.java
package edu.cit.cabasag.barberconnect.feature.social;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

/** Vertical Slice Architecture — Social Feature: Emoji reaction to a post. */
@Data @NoArgsConstructor @AllArgsConstructor
public class Reaction {
    private String reaction_id;
    private String post_id;
    private String user_id;
    private ReactionType type;
    private LocalDateTime createdAt;

    public enum ReactionType { LIKE, LOVE, FIRE, CLAP }
}

File: mobile/app/src/main/java/edu/cit/cabasag/barberconnect/model/Reaction.kt
package edu.cit.cabasag.barberconnect.model

import com.google.gson.annotations.SerializedName

data class Reaction(
    @SerializedName("reaction_id") val reactionId: String? = null,
    @SerializedName("post_id") val postId: String? = null,
    @SerializedName("user_id") val userId: String? = null,
    val type: String? = null,
    val createdAt: String? = null
)
